spec/compiler/normalize/select_spec.cr

```diff
@@ -0,0 +1,68 @@
+require "../../spec_helper"
+
+describe "Normalize: case" do
+  it "normalizes select with call" do
+    assert_expand "select; when foo; body; when bar; baz; end", <<-CODE
+      __temp_1, __temp_2 = ::Channel.select({foo_select_action, bar_select_action})
```

Comment:
Just conceptually, not referring to the existing `Future`, can't `SelectAction` be thought of as a promise/future? `Channel.select` receives a list of promises and returns the first that could compute a value. Following that line of thought, `SelectActon` becomes `Promise` and the suffix becomes `_promise`, and `channel.receive_promise` suddenly reads pretty well to me. Going further the whole concept could be decoupled from `Channel` and `Channel.select` could become `Promise.select`.